mava/networks/sable_network.py

```diff
@@ -0,0 +1,547 @@
+    ) -> chex.Array:
+        """Applies Chunkwise MultiScaleRetention."""
+        ret, updated_hstate = self.retn(
+            key=x, query=x, value=x, hstate=hstate, dones=dones, timestep_id=timestep_id
+        )
+        x = self.ln1(x + ret)
+        output = self.ln2(x + self.ffn(x))
+        return output, updated_hstate
+
+    def recurrent(self, x: chex.Array, hstate: chex.Array, timestep_id: chex.Array) -> chex.Array:
+        """Applies Recurrent MultiScaleRetention."""
+        ret, updated_hstate = self.retn.recurrent(
+            key_n=x, query_n=x, value_n=x, hstate=hstate, timestep_id=timestep_id
+        )
+        x = self.ln1(x + ret)
+        output = self.ln2(x + self.ffn(x))
+        return output, updated_hstate
+
+
+class Encoder(nn.Module):
+    """Multi-block encoder consisting of multiple `EncoderBlock` modules."""
+
+    net_config: SableNetworkConfig
+    memory_config: DictConfig
+    n_agents: int
+
+    def setup(self) -> None:
+        # Initialize the RMSNorm layer normalization
+        self.ln = nn.RMSNorm()
+
+        # Initialize the observation encoder and value head layers
+        self.obs_encoder = nn.Sequential(
+            [
+                nn.RMSNorm(),
+                nn.Dense(
+                    self.net_config.embed_dim, kernel_init=orthogonal(jnp.sqrt(2)), use_bias=False
+                ),
+                nn.gelu,
+            ],
+        )
+        self.head = nn.Sequential(
+            [
+                nn.Dense(self.net_config.embed_dim, kernel_init=orthogonal(jnp.sqrt(2))),
+                nn.gelu,
+                nn.RMSNorm(),
+                nn.Dense(1, kernel_init=orthogonal(0.01)),
+            ],
+        )
+
+        # Initialize the encoder blocks
+        self.blocks = [
+            EncodeBlock(
+                self.net_config,
+                self.memory_config,
+                self.n_agents,
+                name=f"encoder_block_{block_id}",
+            )
+            for block_id in range(self.net_config.n_block)
+        ]
+
+    def __call__(
+        self, obs: chex.Array, hstate: chex.Array, dones: chex.Array, timestep_id: chex.Array
+    ) -> Tuple[chex.Array, chex.Array, chex.Array]:
+        """Apply chunkwise encoding."""
+        # Initialize the updated hidden state
+        updated_hstate = jnp.zeros_like(hstate)
+        # Encode the observation
+        obs_rep = self.obs_encoder(obs)
+
+        # Apply the encoder blocks
+        for i, block in enumerate(self.blocks):
+            # Get the hidden state for the current block
+            hs = hstate[:, :, i]
+            # Apply the chunkwise encoder block
+            obs_rep, hs_new = block(self.ln(obs_rep), hs, dones, timestep_id)
+            updated_hstate = updated_hstate.at[:, :, i].set(hs_new)
+
+        # Compute the value function
+        v_loc = self.head(obs_rep)
+
+        return v_loc, obs_rep, updated_hstate
```

Comment:
v_loc is a bit uninformative
```suggestion
        value = self.head(obs_rep)

        return value, obs_rep, updated_hstate
```